### ISSUE TITLE
[Linux] compiled, gorgeous, but... can not use shift in insert and command mode]

### ISSUE THREAD

**Issue by maxdevjs** [Kethku/neovide#55]
Tried it on [Solus](https://getsol.us/home/).

`Neovide` looks functional, cool, gorgeous and... I have some issue with standard US keyboard layout: can not use `shift` in insert and command mode.

![screenshot-2020-01-27_12:22:15_-03-03:00_1580138535](https://user-images.githubusercontent.com/22229196/73187827-c0487180-4100-11ea-8d72-5093d8974e3c.png)

Any clue?

**Comment by Kethku** [comment]
I don't have a great idea other than the keyboard support is not where it should be. Looks like I need to nuckle down and build a Linux etc box... I may be able to attempt this weekend. In the meantime I would love any help if you (or anyone else) are familiar with Rust

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by maxdevjs** [comment]
Sorry. No idea what Rust is, personally :D 

I saw there are [comments](https://www.reddit.com/r/neovim/comments/euext1/neovide_no_nonsense_neovim_client_in_rust/ffpc99n/) about it on reddit.

**Comment by Kethku** [comment]
Btw It looks like the font you are using isn't fully monospace. That will probably cause you significant problems :(

**Comment by maxdevjs** [comment]
Can you simply embed the font or ask people to install the specific font? Otherwise could be a nightmare...

**Comment by Kethku** [comment]
Font loading works like this today:

1. Look at the config and load the font specified in the vim init scripts via the guifont variable.
2. Load a backup font which should be the default monospace font for your system. I use this function to pick the monospace font: https://docs.rs/font-kit/0.5.0/font_kit/source/trait.Source.html#method.select_best_match
3. Load a manually specified emoji font for each system if it exists.

I have concidered embedding backup font for each of the monospace default and emoji cases, but I haven't done that yet as it would be better to load the already installed fonts.

TLDR: It should already be asking the user for a specific font and using the default for your system.

**Comment by maxdevjs** [comment]
Is it all around here

https://github.com/Kethku/neovide/blob/335045840ba27986524e8a2793fee32aa94bb958/src/renderer/caching_shaper.rs#L36

right? Anyway, I did not find where it reads the config, yet...

**Comment by Kethku** [comment]
Yes thats basically it. The guifont option is propagated from the neovim process via
https://github.com/Kethku/neovide/blob/master/src/bridge/events.rs#L116
which then gets sent to the editor state here
https://github.com/Kethku/neovide/blob/master/src/editor/mod.rs#L253
which is then loaded as the custom font family here
https://github.com/Kethku/neovide/blob/master/src/renderer/caching_shaper.rs#L41

Specifically neovide tries to support a subset of vims guifont option specification. It does not support font weight or italic in that option though currently. https://vim.fandom.com/wiki/Setting_the_font_in_the_GUI

**Comment by maxdevjs** [comment]
Thank you. 

But `neovide` reads somewhere in system vim/nvim configs? If so, where in the code and which physical path?. It is this that I can not find to check against. 

For example, I did not have a `$HOME/.gvimrc` (never used GUI version) file and `fc-match monospace` returns `Hack-Regular.ttf: "Hack" "Regular"` in the system I used to build `neovide`...

**Comment by Kethku** [comment]
neovide isn't in charge of reading the config itself. You set the guifont option in your init.vim script and it gets sent to neovide via an option set event. If you set guifont in your init it won't effect terminal neovim, but will set the font for compliant guis

**Comment by Kethku** [comment]
keybinding wise, this is a repeat of https://github.com/Kethku/neovide/issues/27 which doesn't sound the same, but both have the same root issue. Winit keybinding support is complicated and not great. I have some ideas for how to update it, but its going to take a bit unfortunately

**Comment by maxdevjs** [comment]
Thank you for your time, now it is more clear.

Only way I have been able to set a font, is from command mode. Still, partial functions.

Perhaps you can provide a Windows build and someone can test it with Wine.

Again, great project.

**Comment by Kethku** [comment]
Thanks! Windows build is available in the releases, but it probably won't work given the use of vulkan

**Comment by Kethku** [comment]
note: ginit.vim isn't supported currently. Maybe thats causing you problems?

**Comment by Kethku** [comment]
Closing for now. Feel free to reopen if you're still running into either of the above problems!
